### ISSUE TITLE
Problem with C++ Console input in Webassembly programs REF #7740

### ISSUE THREAD

**Issue by AguwaGeorge** [emscripten-core/emscripten#14887]
https://github.com/emscripten-core/emscripten/issues/7740
This same issue was opened by DragonOsman on Dec. 30, 2018 with the title Input in WebAssembly programs? #7740. However, it was marked as stale and unfortunately closed due to inactivity. But this problem persists!

A simple C++ program like the one below doesn't execute properly when compiled to webassembly using:
`emcc -o3 -g3 --std=c++17 index.cpp -o index.html`.

When run in the browser it keeps triggering the alert() endlessly and stops further execution. Same if run with node.js.
WHAT'S THE SOLUTION??? IT'S ALMOST 3 YEARS SINCE SOMEONE REPORTED THIS ISSUE!!!

```
#include <iostream>
int main() {
	std::cout << "\nEnter your age: ";
	int age {};
	std::cin >> age;
	std::cout << "\nYour age is " << age << ".\n";
	if (age >= 18) {
		std::cout << "Congratulations, you're eligible to vote!";
	} else {
		std::cout << "Sorry, you're still too young to vote.";
	}
	std::cout << std::endl;
	return 0;
}
```

**Comment by sbc100** [comment]
This program works under node if you build with `-sNODERAWFS`.

On the web this also works if you enter you age and then press "cancel" on the next popup.  The reason is that `std::cin >> age;` will read until a newline occurs and the way the input it requests on the web you need to "Cancel" before a newline is sent:

https://github.com/emscripten-core/emscripten/blob/51dcc287bc4337893d329dd4c4cb996c944f9401/src/library_tty.js#L131-L136

This input mechanism on the web is only designed for testing.  If you want to do processing of input from a web app you would want to build some kind if UI in HTML that would receive the input and send it your app.  However, even then reading from stdin doesn't make much sense on the web since that is a blocking operation.

In other words, you probably want to build a bespoke way to get data into and out-of your module rather than use stdin/stdout.  See https://emscripten.org/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.html for various ways to interact with your C++ code.

**Comment by Ender-events** [comment]
You can do something like this

```c++
#ifdef __EMSCRIPTEN__
    age = emscripten_run_script_int("prompt('Enter your age:');");
#else
    std::cin >> age;
#endif
```

**Comment by sbc100** [comment]
I think given the constraints the default behaviour is probably a reasonable compromise.   Any suggestions for improvements welcome of course.
